src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -6936,9 +7014,527 @@ public static void Main(string[] args)
+    const int num3 = 3;
+
+    public static void Main(string[] args)
+    {
+        var fn = delegate(int arg = b1 ? num1 : b2 ? num2 : num3) { return arg; };
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaDefaultLocalConstantExpression()
+        {
+            var source = """
+class Program
+{
+    public static void Main(string[] args)
+    {
+        const int i1 = 1;
+        const int i2 = 2;
+        var func = (int arg = i1 + i2) => arg + 1;
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,19): warning CS0219: The variable 'i1' is assigned but its value is never used
+                //         const int i1 = 1;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i1").WithArguments("i1").WithLocation(5, 19),
+                // (6,19): warning CS0219: The variable 'i2' is assigned but its value is never used
+                //         const int i2 = 2;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i2").WithArguments("i2").WithLocation(6, 19));
+        }
+
+        [Fact]
+        public void AnonymousMethodDefaultLocalConstantExpression()
+        {
+            var source = """
+class Program
+{
+    public static void Main(string[] args)
+    {
+        const int i1 = 1;
+        const int i2 = 2;
+        var func = delegate(int arg = i1 + i2) { return arg + 1; };
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,19): warning CS0219: The variable 'i1' is assigned but its value is never used
+                //         const int i1 = 1;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i1").WithArguments("i1").WithLocation(5, 19),
+                // (6,19): warning CS0219: The variable 'i2' is assigned but its value is never used
+                //         const int i2 = 2;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i2").WithArguments("i2").WithLocation(6, 19));
+        }
+
+        [Fact]
+        public void AsyncLambdaWithDefaultParameters()
+        {
+            var source = """
+using System.Threading.Tasks;
+class Program
+{
+    public static void Main(string[] args)
+    {
+        var lam = async (int delay = 10) => await Task.Delay(delay);
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void StaticLambdaWithDefaultParameters()
+        {
+            var source = """
+class Program 
+{
+    public static void Main()
+    {
+        var lam = static (string s = "str") => s;
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaWithDefaultParamsAndRefOutModifiers()
```

Comment:
There's no params in this test (and the next one)?

Reply:
Oh, I see, you using it as a shorthand. Consider spelling out Parameters, as `params` is a C# keyword we'll be testing at some point.